### PR TITLE
GitHub followers

### PR DESCRIPTION
**Pull request by elguapogordo**
add follower tests and function
fix repositories tests and function broken by follower additions
minor rubocop fixes

### CODE REVIEW

File: app/apis/github_v3_api.rb [hunk 1]
```diff
@@ -3,25 +3,27 @@ def initialize(api_key = nil)
...
   def connect
     Faraday.new('https://api.github.com') do |req|
       req.headers['Authorization'] = @api_key
     end
   end
 
-  def repo_response
-    connect.get('/user/repos')
+  def response(endpoint)
+    connect.get("/user/#{endpoint}")
```
Comment (elguapogordo) [inline app/apis/github_v3_api.rb:23]: consolidated `repo_response` and `followers_response` into one method here

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,6 +1,9 @@
 class UsersController < ApplicationController
   def show
-    @repos = gather_repo_objects if github_token_present?
+    return unless github_token_present?
+
+    @repos = RepositoryCollection.new(current_user).return_collection
+    @followers = FollowersCollection.new(current_user).return_collection
   end
```
Comment (elguapogordo) [inline app/controllers/users_controller.rb:7]: i still like the `if` statement better. unfortunately rubocop does not.

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -16,6 +16,12 @@
       <% @repos.each_with_index do |repo, index| %>
         <p class="repo"><%= link_to repo.name, repo.url, id:"#repo-#{index}" %></p>
       <% end %>
+      <section class='followers'>
+        <h3>Followers</h3>
+        <% @followers.each_with_index do |follower, index| %>
+          <p class='follower'><%= link_to follower.name, follower.url, id:"#follower-#{index}" %></p>
+        <% end %>
```
Comment (elguapogordo) [inline app/views/users/show.html.erb:23]: would still like to see an edgecase `if` for user with no followers

**Review by cmpprg — APPROVED** [review]
Everything looks good, we should rename the collections to Factories

**Pull request merged**